app/views/govuk_publishing_components/components/_search.html.erb

```diff
@@ -13,6 +14,13 @@
   data_attributes ||= {}
   data_attributes[:module] = 'gem-track-click'
 
+  label_classes = []
+  if (["xl", "l", "m", "s"].include?(label_size))
```

Comment:
There's a method in the shared_helper that does something very similar to this... it might be worth considering if it could be reused? https://github.com/alphagov/govuk_publishing_components/blob/master/lib/govuk_publishing_components/presenters/shared_helper.rb#L34

Reply:
The most I think that this can be abstracted out is a method to have once place to check if the size is correct, since the sizes are appended to a number of different classes in the fieldset, input, label, radio, and select components. Otherwise we'd have a "shared" helper for each component - one for `govuk-label--X`, one for `govuk-fieldset--X` etc - which seems slightly pointless.

I've added a helper method and updated the components, including the search component, to use it: https://github.com/alphagov/govuk_publishing_components/blob/06f063931cd756367c9fe88ff0cc3477a871fc92/lib/govuk_publishing_components/presenters/shared_helper.rb#L34-L36

Reply:
That looks good.